contrib/inventory/netbox.py

```diff
@@ -0,0 +1,344 @@
+        self.list = script_args.list
+        self.host = script_args.host
+
+        # Script configuration.
+        main_config_key = "netbox"
+        script_config = script_config_data.get(main_config_key)
+        if script_config:
+            self.api_url = script_config["main"].get('api_url')
+            self.group_by = script_config.setdefault("group_by", {})
+            self.hosts_vars = script_config.setdefault("hosts_vars", {})
+        else:
+            sys.exit("The key '%s' is not found in config file." % main_config_key)
+
+        # Get value based on key.
+        self.key_map = {
+            "default": "name",
+            "general": "name",
+            "custom": "value",
+            "ip": "address"
+        }
+
+    @staticmethod
+    def get_hosts_list(api_url, specific_host=None):
+        """Retrieves hosts list from netbox API.
+
+        Returns:
+            A list of all hosts from netbox API.
+        """
+
+        if not api_url:
+            sys.exit("Please check API URL in script configuration file.")
+
+        if specific_host:
+            data_source = "{}?name={}".format(api_url, specific_host)
+        else:
+            data_source = api_url
+
+        hosts_list = requests.get(data_source).json()
+        return hosts_list
+
+    @staticmethod
+    def add_host_to_group(server_name, group_value, inventory_dict):
+        """Add a host to a single group.
+
+        It checks if host in a group and adds the host to that group.
+        The group will be added if it's not in the inventory.
+
+        Args:
+            server_name: String, the server that will be added to a group.
+            group_value: String, name that will be used as a group in the inventory.
+            inventory_dict: Dict, the inventory which will be updated.
+
+        Returns:
+            This function doesn't return, it updates the dict in place.
+        """
+
+        # The value could be None/null.
+        if group_value:
+            # If the group not in the inventory it will be add.
+            if group_value not in inventory_dict:
+                inventory_dict.update({group_value: []})
+
+            # If the host not in the group it will be add.
+            if server_name not in inventory_dict[group_value]:
+                inventory_dict[group_value].append(server_name)
+
+    def add_host_to_inventory(self, groups_categories, inventory_dict, host_data):
+        """Add a host to its groups.
+
+        It checks if host in the groups and adds the host to these groups.
+        The groups are defined in this inventory script config file.
+
+        Args:
+            groups_categories: Dict, it has a categories of groups that will be
+                used as Ansible inventory groups.
+            inventory_dict: Dict, which is Ansible inventory.
+            host_data: Dict, it has the host data that will be added to inventory.
+
+        Returns:
+            The same dict "inventory_dict" after update.
```

Comment:
Noticed inconsistency between ``add_host_to_inventory`` and ``add_host_to_group`` with regard to return value.